composables/transaction/mintCollection/useNewCollectionId.ts

```diff
@@ -1,32 +1,20 @@
-import { getRandomValues } from '@/components/unique/utils'
-import { mapToId } from '@/utils/mappers'
-import { unwrapSafe } from '@/utils/uniquery'
-
 export function useNewCollectionId() {
+  const { apiInstance } = useApi()
   const newCollectionId = ref<number>()
-  const randomNumbers = getRandomValues(10).map(String)
 
-  const { data, error, loading, refetch } = useGraphql({
-    queryName: 'existingCollectionList',
-    variables: {
-      ids: randomNumbers,
-    },
-  })
+  const getCollectionId = async () => {
+    const api = await apiInstance.value
+    const nextCollectionId = (
+      await api.query.nfts.nextCollectionId()
+    ).value.toNumber()
+    newCollectionId.value = nextCollectionId
+  }
 
-  watch(data, () => {
-    if (data.value) {
-      const collectionEntities = data.value.collectionEntities
-      const collectionList = unwrapSafe(collectionEntities)
-      const existingIds = collectionList.map(mapToId)
-      const newId = randomNumbers.find((id) => !existingIds.includes(id))
-      newCollectionId.value = Number(newId)
-    }
+  getCollectionId().catch((e) => {
+    console.error('Error:', e)
   })
 
   return {
     newCollectionId,
-    loading,
-    error,
-    refetch,
```

Comment:
Do you know that we need this for BSX and STT?

Reply:
the entire function looked like a placeholder to me
you generate 10 random numbers to fetch from the db and then return the first one that doesn't exist already in the db

 but what if you happen to generate 10 numbers all already existing?

Reply:
anyhow ill restore it and use subscribe in another composable to get the id for STMN collection
Thank you :smile:

Reply:
Because, i dont know which collection number should i use so very random 32-bit integers have enough entropy. 

> all already existing?

Overoptimization vs working code. 
Worst case that could happen now is we tell user to try again